src/eva/vision/utils/io/nifti.py

```diff
@@ -28,27 +29,39 @@ def read_nifti_slice(
     """
     _utils.check_file(path)
     image_data = nib.load(path)  # type: ignore
-    image_slice = image_data.slicer[:, :, slice_index : slice_index + 1]  # type: ignore
-    image_array = image_slice.get_fdata()
+    if slice_index is not None:
+        image_data = image_data.slicer[:, :, slice_index : slice_index + 1]  # type: ignore
+    image_array = image_data.get_fdata()  # type: ignore
     if use_storage_dtype:
         image_array = image_array.astype(image_data.get_data_dtype())  # type: ignore
     return image_array
 
 
-def fetch_total_nifti_slices(path: str) -> int:
-    """Fetches the total slides of a NIfTI image file.
+def save_array_as_nifti(array: npt.ArrayLike, filename: str) -> None:
```

Comment:
When should we use `np.ndarray` vs `npt.ArrayLike` as type hints? (asking because in many other places we use `np.ndarray`)

Reply:
the correct one - according to `pyright` - is the `npt.ArrayLike`

its interesting that in this case, it complained and worked only with `npt.ArrayLike`. So ideally I would say we should use the `npt.ArrayLike` everywhere - actually we should open an issue about that

Reply:
yeah let's open an issue for this :)